### PR TITLE
ISPN-9666 Agroal JDBC Connection pool added

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-9666
https://issues.jboss.org/browse/ISPN-8087

`sl4fj-jboss-logmanager` dependency also removed as it was only required by HikariCP.

### CODE REVIEW

File: persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java [hunk 1]
```diff
@@ -47,15 +43,38 @@ public void start(ConnectionFactoryConfiguration config, ClassLoader classLoader
...
+
+         String propsFile = poolConfig.propertyFile();
+         if (propsFile != null) {
+            dataSource = AgroalDataSource.from(new AgroalPropertiesReader(PROPERTIES_PREFIX).readProperties(propsFile));
+         } else {
+            // Default Agroal configuration with a fixed size connection-pool and metrics disabled
+            String password = poolConfig.password() != null ? poolConfig.password() : "";
+            AgroalDataSourceConfigurationSupplier configuration = new AgroalDataSourceConfigurationSupplier()
+                  .connectionPoolConfiguration(cp -> cp
+                        .minSize(10) // Same as maxSize so we have a fixed size connection pool to maximize responsiveness
```
Comment (tristantarrant) [inline persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java]: I think this is unnecessarily aggressive. If the user wants tuning they can use the props file.

**Comment by tristantarrant** [comment]
LGTM, one minor comment.

File: persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java (hunk 1 above)
Comment (ryanemerson) [reply persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java]: Yeah fair enough, I was just re-using the HikariCP default settings there. Updated.

**Comment by galderz** [comment]
Looks good as well, integrated.

**Pull request merged**